internal/handlers/pg/msg_find.go

```diff
@@ -143,3 +193,66 @@ func (h *Handler) fetchAndFilterDocs(ctx context.Context, tx pgx.Tx, sqlParam *p
 		resDocs = append(resDocs, doc)
 	}
 }
+
+// getFirstBatchAndIterator fetches documents from the database and filters them using the provided sqlParam.Filter.
+func (h *Handler) getFirstBatchAndIterator(ctx context.Context, tx pgx.Tx, sqlParam *pgdb.SQLParam) (
+	[]*types.Document, iterator.Interface[int, *types.Document], error,
+) {
+	iter, err := pgdb.GetDocuments(ctx, tx, sqlParam)
```

Comment:
Okay so this transaction can live long time potentially until all document from getMore is fetched. In terms of pg, this is not ideal.

And I suppose, it's here because our iterator assumes underling transaction exists.

Let me think on it. :thinking:

Reply:
That's fine: that transaction doesn't modifies data, and PostgreSQL's MVCC ensures that this transaction is relatively lightweight. It still consumes a connection, but that's true even without transaction, just with a cursor.

We still should be careful about closing both iterator and transaction, of course.

Reply:
Good to know. Yes we need to be careful about closing them.